troubleshooting.md

```diff
@@ -0,0 +1,551 @@
+```
+Once logged in to this VM, please check if all the pods(dnsmasq,mariadb,httpd, ironic pods etc) are in running state. 
+
+To check the logs of all the containers , please the below commands:
+
+```
+sudo podman logs ironic-api
+```
+You will get the error messages if the master nodes are not `powered on` by inspecting the ironic pods. 
+
+Once the ironic poewers on the master nodes, verify if RHCOS images are pushed to the cluster, once again inspect the logs of httpd and ironic pods logs from Bootstrap VM. 
+
+## BootstrapVM cannot download RHCOS image
+
+After the Bootstrap VM goes up comes a phase where podman raises 2 containers:
+
+```
+ipa-downloader
+coreos-downloader
+```
+
+These both containers will download resources from (in our case) the jump / bastion host since its a disconnected install.This is the relevant section where is this step performed, modify it in order to let the provisioner to access to the resources:
+
+```
+bootstrapOSImage: http://[<IPv4/Ipv6-IP-Registry]/rhcos-43.81.202001142154.0-qemu.x86_64.qcow2.gz?sha256=9d999f55ff1d44f7ed7c106508e5deecd04dc3c06095d34d36bf1cd127837e0c
+clusterOSImage: http://[<IPv4/Ipv6-IP-Registry>]/rhcos-43.81.202001142154.0-openstack.x86_64.qcow2.gz?sha256=a1bda656fa0892f7b936fdc6b6a6086bddaed5dafacedcd7a1e811abb78fe3b0
+```
+
+##Check the status of these containers using 
+
+```
+#podman logs -f ipa-downloader
+
+#podman logs -f coreos-downloader
+
+```
+After the installation is complete copy the config file to the `~/.kube/config` dir to interact with the cluster
+
+```
+cp /install-dir/ocp/auth/kubeconfig  ~/.kube/config
+```
+Or 
+
+```
+export KUBECONFIG=/install-dir/ocp/auth/kubeconfig
+```
+
+Verify if all the control plane nodes are up and running 
+
+```
+oc get nodes
+NAME                                         STATUS   ROLES           AGE    VERSION
+master-0.cloud.lab.eng.bos.redhat.com   Ready    master,worker   4h   v1.16.2
+master-1.cloud.lab.eng.bos.redhat.com   Ready    master,worker   4h   v1.16.2
+master-2.cloud.lab.eng.bos.redhat.com   Ready    master,worker   4h   v1.16.2
+```
+
+There might be a scenario where the nodes can be stuck in `NotReady` state. To check the exact errors, login to the nodes and check for logs.
+
+```
+#ssh core@master-0 
+#crictl ps (look for the error in pod status if any)
+#crictl logs <pod-in-error-state>
+```
+Check for pod errors if any in the cluster in any namespaces
+
+```
+oc get namespaces 
+For Example: oc get pods -n openshift-kube-apiserver
+```
+
+## Bootstrap VM cannot boot up my nodes
+
+This is a usual problem, maybe we add a typo on the install-config.yaml file or just the privileges are not right on my user. In the worst case you have a problem with your network prerequisites regarding routing and accesses to iDrac from Ironic. 
+To check the problems here we have 3 basic containers all regarding ironic:
+    * Ironic-api
+    * Ironic-conductor
+    * Ironic-inspector
+
```

Comment:
What about a checklist of what to check when bootstrap VM cannot boot up the nodes

Reply:
This above paragraph is borrowed from installer troubleshooting document, I will extend this list.